app/views/RoomActionsView/index.js

```diff
@@ -412,6 +414,79 @@ class RoomActionsView extends React.Component {
 		);
 	}
 
+	handleLeaveTeam = async(selected) => {
+		const { room } = this.state;
+		try {
+			const { navigation, isMasterDetail } = this.props;
+			const result = await RocketChat.leaveTeam({ teamName: room.name });
+
+			if (selected) {
+				try {
+					selected.map(item => RocketChat.leaveRoom(item.rid, item.t));
+				} catch (e) {
+					log(e);
+				}
+			}
+			if (result.success) {
+				if (isMasterDetail) {
+					navigation.popToTop();
+				} else {
+					navigation.navigate('RoomsListView');
+				}
+			}
+		} catch (e) {
+			log(e);
+			Alert.alert(
+				I18n.t('Cannot_leave'),
+				e.data?.error ? I18n.t(e.data.error) : I18n.t('There_was_an_error_while_action', { action: I18n.t('leaving_team') }),
+				[
+					{
+						text: 'OK',
+						style: 'cancel'
+					}
+				]
+			);
+		}
+	}
+
+	leaveTeam = async() => {
+		const { room } = this.state;
+		const { navigation } = this.props;
+
+		try {
+			const db = database.active;
+			const subCollection = db.get('subscriptions');
+			const teamChannels = await subCollection.query(
+				Q.where('team_id', room.teamId),
+				Q.where('team_main', null)
+			);
+
+			if (teamChannels.length) {
+				navigation.navigate('SelectListView', {
+					title: 'Leave_Team', room, data: teamChannels, subtitle: 'Select_Teams', nextAction: data => this.handleLeaveTeam(data)
```

Comment:
`handleLeaveTeam` param is too vague.
Here it receives `data`, which seems to be an Array.
Below in the code, it receives `room.name`, which is a String.

Reply:
Fixed